src/webhook/upgrade_operator/fetch_upgradeconfig.py

```diff
@@ -1,19 +1,25 @@
 import os
 from kubernetes import client, config
 from kubernetes.client.rest import ApiException
+from kubernetes.config.config_exception import ConfigException
 
 # Use KUBECONFIG from pod else consume from local ~/.kube/config
-if 'KUBERNETES_PORT' in os.environ:
-    config.load_incluster_config()
-else:
-    config.load_kube_config()
+try:
+    if 'KUBERNETES_PORT' in os.environ:
```

Comment:
Sure, have fixed this logic to separate out the auth from within pod and outside pod. Having auth using ~/.kube/config is as such not to be part of the code since it is only meant to run within the pod but to locally run the script, have added this part for now. It's optional part.